bin/tfl/import_categories

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env perl
+
+use warnings;
+use v5.14;
+use utf8;
+
+BEGIN {
+    use File::Basename qw(dirname);
+    use File::Spec;
+    my $d = dirname(File::Spec->rel2abs($0));
+    require "$d/../../setenv.pl";
+}
+
+use FixMyStreet;
+use FixMyStreet::DB;
+use Getopt::Long::Descriptive;
+use JSON::MaybeXS;
+use Path::Tiny;
+
+my ($opt, $usage) = describe_options(
+    '%c %o',
+    [ 'commit', "Actually commit changes to the database" ],
+    [ 'help', "print usage message and exit", { shortcircuit => 1 } ],
+);
+print($usage->text), exit if $opt->help;
+
+die "Usage: $0 <path/to/categories.json>" unless -f $ARGV[0];
+
+my $db;
+END {
+    if ($db) {
+        $opt->commit ? $db->txn_commit : $db->txn_rollback;
+    }
+}
+
+$db = FixMyStreet::DB->schema->storage;
+$db->txn_begin;
+if (!$opt->commit) {
+    say "NOT COMMITTING TO DATABASE";
+}
+
+my $config = decode_json(path($ARGV[0])->slurp_utf8);
+
+my $body = FixMyStreet::DB->resultset('Body')->find({ name => 'TfL' });
+
+say "Found body " . $body->name;
+
+my $groups = $config->{groups};
+for my $group (keys %$groups) {
+    my $cats = $groups->{$group};
+    for my $cat (@$cats) {
+        my $child_cat = FixMyStreet::DB->resultset("Contact")->find_or_new({
+            body => $body,
+            category => $cat->{category}
+        });
+        $child_cat->email($cat->{email});
+        $child_cat->state('confirmed');
+        $child_cat->editor($0);
+        $child_cat->whenedited(\'current_timestamp');
+        $child_cat->note($child_cat->in_storage ? 'Updated by import_categories' : 'Created by import_categories');
+        my $groups = $child_cat->groups;
```

Comment:
There aren't any child groups in the example data (in case this code isn't needed any more).

Reply:
The `Other` category appears in multiple groups, so might already have had the group set by an earlier iteration.

Reply:
Ah yes In which case, only slight annoyance is it creating many history items for that.

Reply:
Yeah, bit annoying, but no big deal as the script should only be run the once.